router/internal/eventorder/eventorder.go

```diff
@@ -0,0 +1,218 @@
+package eventorder
```

Comment:
This package contains the new event order algorithm implementation. Although it is not used by router yet, it is still tested under similar conditions in `eventorder_simulation_test.go`